libtelemetry/src/main/java/com/mapbox/android/telemetry/NavigationMetadata.java

```diff
@@ -45,17 +45,25 @@
   private Boolean batteryPluggedIn;
   private Integer batteryLevel;
   private String connectivity;
+  private String tripIdentifier;
+  private Integer legIndex;
```

Comment:
What are some benefits of using boxed type here as opposed to primitive?

Reply:
I see what happened here. While attempting to keep consistency, I mimicked `Integer` we use for the `null` checks when serializing. Will remove those that do not require the null check.